src/porepy/models/fluid_property_library.py

```diff
@@ -273,6 +275,187 @@ def interface_mobility_discretization(
         """
         return pp.ad.UpwindCouplingAd(self.mobility_keyword, interfaces)
 
+    def total_mass_mobility(self, domains: pp.SubdomainsOrBoundaries) -> pp.ad.Operator:
+        r"""Total mass mobility of the fluid mixture is given by
+
+        .. math::
+
+                \sum_j \frac{\rho_j k_r(s_j)}{\mu_j}.
+
+        Used as a non-linear part of the diffusive tensor in the (total) mass balance
+        equation.
+
+        Note:
+            In the single-phase, single-component case, this is reduced to
+            :math:`\frac{\rho}{\mu}`.
+
+        Parameters:
+            domains: A list of subdomains or boundary grids.
+
+        Returns:
+            Above expression in operator form.
+
+        """
+        name = "total_mass_mobility"
+        mobility = pp.ad.sum_operator_list(
+            [
+                phase.density(domains) * self.phase_mobility(phase, domains)
+                for phase in self.fluid.phases
+            ],
+            name,
+        )
+        return mobility
+
+    def phase_mobility(
+        self, phase: pp.Phase, domains: pp.SubdomainsOrBoundaries
+    ) -> pp.ad.Operator:
+        r"""Returns the mobility of a phase :math:`j`
+
+        .. math::
+
+            \frac{k_r(s_j)}{\mu_j}.
+
+        Notes:
+            For the single-phase case it returns simply :math:`\frac{1}{\mu}`.
+
+        Important:
+            Contrary to all other mobility methods implemented here, this one does not
+            contain any mass term, it is a volumetric term. This is the term commonly
+            denoted 'mobility in the literature.
+
+        Parameters:
+            phase: A phase in the fluid mixture.
+            domains: A sequence of subdomains or boundary grids.
+
+        Returns:
+            Above expression in operator form.
+
+        """
+        # Distinguish between single-phase case and multi-phase case: Usage of rel-perm
+        # makes this class compatible with single-phase models, without requiring some
+        # rel-perm mixin.
+        if self.fluid.num_phases > 1:
+            mobility = self.relative_permeability(phase, domains) / phase.viscosity(
+                domains
+            )
+        else:
+            assert phase == self.fluid.reference_phase
+            mobility = phase.viscosity(domains) ** pp.ad.Scalar(-1.0)
+        mobility.set_name(f"phase_mobility_{phase.name}")
+        return mobility
+
+    def component_mass_mobility(
```

Comment:
We should add tests of a these methods.

Reply:
All of them are tested the same way the previous `mobility` method was tested in `test_fluid_mass_balance`, including a check for correct values.